### DISCUSSION TITLE
Add nutritions to ingredients

### DISCUSSION THREAD

**Discussion by Mac0501** [mealie-recipes/mealie#4357] [Feature Request]
### First Check

- [X] I used the GitHub search to find a similar requests and didn't find it.
- [X] Checked the [tasks tagged](https://github.com/mealie-recipes/mealie/issues?q=is%3Aissue+is%3Aopen+label%3Atask+) issues and verified my feature is not covered

### Please provide a concise description of the problem that would be addressed by this feature.

It's a lot of work every time to calculate what nutritional values ​​the dish has for a serving.
Therefore, it would be easier to enter the nutritional values ​​once for each ingredient and then have the nutritional values ​​for a recipe calculated automatically when creating a recipe.

### Please provide a concise description of the feature that would resolve your issue.

In the list where you can manage the meta data for foods, a nutritional values ​​function should be added, which opens in a pop-up where you can set nutritional values. If necessary, you can use this pop-up to completely set up a food
It would also be nice if more nutritional values ​​could be entered as a reference you could use Google
sry for the German screenshot
![image](https://github.com/user-attachments/assets/139ba62d-6d6b-4c38-96fc-d61c477dfa04)

### Please consider and list out some caveats or tradeoffs made in your design decision

Of course, it shouldn't be mandatory to enter this information

### Additional Information

- [ ] If this is accepted I'm willing to submit a PR to provide this feature
- [ ] If this is accepted I'm willing to help maintain this feature
- [ ] I'm willing to sponsor/pay a developer to do this work

**Comment by michael-genson** [comment]
One of the main challenges of this feature is units/conversions. We would need to somehow determine the nutritional value of a quantity + unit + food.

I think this would be a valuable feature, it's just a very big undertaking.

**Reply by Mac0501** [reply]
yes, I thought about it too, for example: To assign a value to teaspoons or tablespoons or cups etc in order to then create a uniform conversion. It would also make sense to state the nutritional values ​​in 100 g or to include a conversion there and state whether it is 100g or 10g for the set nutrition of a food.

I would be willing to make a pull request for this feature, but firstly I don't have much time because I'm in my penultimate semester and secondly, I'm not that familiar with coding guidelines.

**Reply by michael-genson** [reply]
Yup, for sure. There are a few discussions around this/similar:
- https://github.com/mealie-recipes/mealie/discussions/1659
- https://github.com/mealie-recipes/mealie/discussions/2748
- https://github.com/mealie-recipes/mealie/discussions/2766
- https://github.com/mealie-recipes/mealie/discussions/2996

**Comment by gilbrotheraway** [comment]
https://www.reddit.com/r/selfhosted/comments/18wy55c/comment/kgehcpd/

does not sound so complicated 

you're losing a lot of potential users 
like people who actually care about their diet

**Reply by Kuchenpirat** [reply]
This is glancing over a lot regarding our setup if you go into a bit more detail. 

Units: 
  - Units are currently just strings, so we would need to implement a few base Units and add relations for every unit to it's base unit. 
    - This means that users would have to provide a conversion to a base unit for every new unit they create that is not seeded. Conversion to base unit is not always possible for a lot of units people intuitively use (e.g. clove, bunch and many more)
    - What do we do when there is no unit provided (e.g. 5 tomatoes). We would first have to figure out how big/heavy the average item is and use that adding another komplication. But to be fair they have that implemented for a lot of foods (in a kinda bad way).

Foods: 
  - Finding a food in that database is not as straight foreward as one might think. Just search for potato or spinach. It usually responds with a list of products to choose from. So we would first have to match each ingredient to a Database ID to be certain. 
  - What do we do about foods that are not included in the DB 
  - Mealie gets used all over the world. This means that foods are probably more often than not, not in english. So we would need to translate each food to an english cunterpart and then seach the DB. 

While i agree that this would be a super cool featurer i don't know i we would be able with the hurdles above. Easiest implementation would probably be via our OpenAi integration which would come with its own set of problems. 

Responding in a lot of detail to be able to copy that elsewhere, doing automatic stuff with units and foods comes up quite often.
